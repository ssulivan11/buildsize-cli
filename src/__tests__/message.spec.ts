const message = require('../message');

jest.mock('shelljs', () => ({
  exit: () => {}
}));

describe('message return', () => {
  test('success', () => {
    console.log = jest.fn();
    message('success', 'mock message');
    expect(console.log).toMatchSnapshot();
  });

  test('exit', () => {
    console.log = jest.fn();
    message('exit', 'mock message');
    expect(console.log).toMatchSnapshot();
  });

  test('failure', () => {
    console.log = jest.fn();
    message('failure', 'mock message');
    expect(console.log).toMatchSnapshot();
  });

  test('warning', () => {
    console.log = jest.fn();
    message('warning', 'mock message');
    expect(console.log).toMatchSnapshot();
  });

  test('complete', () => {
    console.log = jest.fn();
    message('complete', 'mock message');
    expect(console.log).toMatchSnapshot();
  });
});

export {};
