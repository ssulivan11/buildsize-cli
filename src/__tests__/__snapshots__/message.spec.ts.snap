// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`message return complete 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "

ðŸš€ mock message
",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`message return exit 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "

ðŸ’© mock message
",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`message return failure 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "[101m[30m[30m
 failure [0m[0m[0m",
      "[31mmock message[0m",
      undefined,
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`message return success 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "[102m[30m[30m
 success [0m[0m[0m",
      "[32mmock message[0m",
      undefined,
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`message return warning 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "[103m[30m[30m
 warning [0m[0m[0m",
      "[33mmock message[0m",
      undefined,
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
